### PR TITLE
Feature/playerscrubber

### PR DESCRIPTION
**Pull request by mrjamesjarvis**
Music scrubber + seek-to logic implementation

### CODE REVIEW

File: src/components/Core/Player/Player.jsx [hunk 1]
```diff
@@ -132,18 +139,38 @@ export default class Player extends React.Component {
...
-  renderProgress() {
-    const t = this.state.playbackTime;
-    if (!t || t.currentPlaybackDuration === 0) {
-      return (<div className={styles["progress-bar"]}><div/></div>);
+  handleSeek(percent, duration) {
+    const time = this.percentToTime(percent, duration);
+    this.changePlaybackTime(time);
+    this._scrubToTime(time);
+  }
+  scrubToTime(time) {
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.jsx]: \n

File: src/components/Core/Player/Player.jsx [hunk 2]
```diff
@@ -132,18 +139,38 @@ export default class Player extends React.Component {
...
+  }
+
+  timeToPercent(time, duration) {
+    if(duration === 0) {
+      return 0; // For some reason would call this
     }
-    const percent = (t.currentPlaybackTime * 100) / t.currentPlaybackDuration;
+    return Math.floor((time * 100) / duration);
+  }
+  percentToTime(percent, duration) {
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.jsx]: \n

File: src/components/Core/Player/Player.scss [hunk 3]
```diff
@@ -48,23 +48,34 @@
...
+        -webkit-appearance: none;
+    }
+    
+    
+    &::-webkit-slider-thumb {
+      height: 10px;
+      width: 5px;
+      border: 0;
       background: rgb(100, 100, 100);
+      // border: 1px solid #777;
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.scss]: Remove commented code

File: src/components/Core/Player/Player.scss [hunk 4]
```diff
@@ -48,23 +48,34 @@
...
-        content: '';
-        background: rgb(100, 100, 100);
-        position: absolute;
-        right: -1px;
-        top: -2px;
-        height: 8px;
-        width: 3px;
-        border-radius: 2px;
-      }
+      // box-shadow: 0 0 1px 0px rgba(0,0,0,0.1);
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.scss]: Remove commented code

File: src/components/Core/Player/Player.scss [hunk 5]
```diff
@@ -48,23 +48,34 @@
...
-      width: 0%;
-      height: 100%;
+    margin: auto;
+    outline: none;
+    padding: 0;
+    // width: 640px;
+    background-color: #dedede;
+    background-image: -webkit-gradient(linear, 50% 0%, 50% 100%, color-stop(0%, #fe2851), color-stop(100%, #fe2851));
+    background-repeat: no-repeat;
+    // border-radius: 10px;
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.scss]: Remove commented code

File: src/components/Core/Player/Player.scss [hunk 6]
```diff
@@ -48,23 +48,34 @@
     margin-top: 10px;
     border-radius: 2px;
 
-    > div {
-      width: 0%;
-      height: 100%;
+    margin: auto;
+    outline: none;
+    padding: 0;
+    // width: 640px;
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.scss]: Remove commented code

File: src/components/Core/Player/Player.jsx [hunk 7]
```diff
@@ -132,18 +139,40 @@ export default class Player extends React.Component {
...
-    const percent = (t.currentPlaybackTime * 100) / t.currentPlaybackDuration;
+    return Math.floor((time * 100) / duration);
+  }
+
+  percentToTime(percent, duration) {
+    return Math.floor((percent * duration) / 100);
+  }
+
+  renderProgress() {
+    const duration = Math.round(this.state.nowPlayingItem.playbackDuration/1000);
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.jsx]: Use object deconstruction

File: src/components/Core/Player/Player.jsx [hunk 8]
```diff
@@ -24,6 +25,9 @@ export default class Player extends React.Component {
 
     this.handlePrevious = this.handlePrevious.bind(this);
     this.handleNext = this.handleNext.bind(this);
+    this.handleSeek = this.handleSeek.bind(this);
+
+    this._scrubToTime = _.debounce(this.scrubToTime, 100).bind(this);
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.jsx]: No `_`

File: src/components/Core/Player/Player.jsx [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import styles from './Player.scss'
 import {artworkForMediaItem} from "../common/Utils";
+import _ from 'lodash';
```
Comment (raphaelvigee) [inline src/components/Core/Player/Player.jsx]: Lets import only the `debounce` function of lodash

File: src/components/Core/Player/Player.scss (hunk 3 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.scss]: Resolved.

File: src/components/Core/Player/Player.scss (hunk 4 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.scss]: Resolved.

File: src/components/Core/Player/Player.scss (hunk 5 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.scss]: Resolved.

File: src/components/Core/Player/Player.scss (hunk 6 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.scss]: Resolved.

File: src/components/Core/Player/Player.jsx (hunk 7 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.jsx]: Resolved.

File: src/components/Core/Player/Player.jsx (hunk 8 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.jsx]: Resolved.

File: src/components/Core/Player/Player.jsx (hunk 9 above)
Comment (BrychanOdlum) [reply src/components/Core/Player/Player.jsx]: Resolved.

**Pull request merged**